### ISSUE TITLE
Gl200TextProtocolDecoder.java decodes zero ODOMETER for multi location messages

### ISSUE THREAD

**Issue by ckrey** [tananaev/traccar#3805]
If a `GTFRI` message contains more than one location, GL200TextProtocolDecoder.java
decodes zero for ODOMETER for all but the last location:

<img width="901" alt="bildschirmfoto 2018-03-12 um 06 47 09" src="https://user-images.githubusercontent.com/4027289/37268098-7acd4d5a-25c4-11e8-93d1-c108a75d6938.png">

According to the documentation the mileage is not part of the repeating group per position, but is contained in the message only once:

<img width="356" alt="bildschirmfoto 2018-03-12 um 07 08 22" src="https://user-images.githubusercontent.com/4027289/37268106-8656eeba-25c4-11e8-98ec-f69d6ce79a2a.png">

tracker-server.log
```
2018-03-11 14:46:08 DEBUG: [723A9FE1: 5004 < 127.0.0.1] HEX: 2b524553503a47544652492c3338303730312c3836393630363032303234363930302c2c2c35312c312c312c342e322c3333322c33372e322c362e3739323136352c35312e3231393937342c32303138303331313133343630372c303236322c303030332c384342342c344532342c30302c3434322e382c2c2c2c39392c3232303130302c2c2c2c32303138303331313133343630382c3130414124
2018-03-11 14:46:08  INFO: [723A9FE1] id: 869606020246900, time: 2018-03-11 14:46:07, lat: 51.21997, lon: 6.79217, speed: 2.3, course: 332.0
...
2018-03-11 14:47:26 DEBUG: [723A9FE1: 5004 < 127.0.0.1] HEX: 2b524553503a47544652492c3338303730312c3836393630363032303234363930302c2c2c35302c332c312c302e302c3331312c33352e382c362e3739323033312c35312e3232303030392c32303138303331313133343632322c303236322c303030332c384342342c344532342c30302c312c31362e322c3239342c34362e302c362e3738393932302c35312e3232303630352c32303138303331313133343732322c303236322c303030332c384342342c344532342c30302c312c32312e312c3234362c34372e332c362e3738393732362c35312e3232303631342c32303138303331313133343732352c303236322c303030332c384342342c344532342c30302c3434332e302c2c2c2c3130302c3232303130302c2c2c2c32303138303331313133343732362c3130414224
2018-03-11 14:47:26  INFO: [723A9FE1] id: 869606020246900, time: 2018-03-11 14:46:22, lat: 51.22001, lon: 6.79203, speed: 0.0, course: 311.0
2018-03-11 14:47:26  INFO: [723A9FE1] id: 869606020246900, time: 2018-03-11 14:47:22, lat: 51.22061, lon: 6.78992, speed: 8.7, course: 294.0
2018-03-11 14:47:26  INFO: [723A9FE1] id: 869606020246900, time: 2018-03-11 14:47:25, lat: 51.22061, lon: 6.78973, speed: 11.4, course: 246.0
```

Gl200TextProtocolDecoder.java
```
 470     private void decodeLocation(Position position, Parser parser) {
 ...
 499         position.set(Position.KEY_ODOMETER, parser.nextDouble(0) * 1000);
 500     }
```

**Comment by tananaev** [comment]
Are you sure it's zero in the database? Or it just shows zero in the user interface because value is not present?

**Comment by ckrey** [comment]
yep:

```
mysql> select devicetime, attributes from positions where deviceid = 49 and devicetime > "2018-03-11 14:46:00" and devicetime < "2018-03-11 14:48:00";
+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
| devicetime          | attributes                                                                                                                                                    |
+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
| 2018-03-11 14:46:08 | {"hdop":1,"odometer":442800.0,"batteryLevel":99,"ignition":true,"input":1,"output":0,"type":"FRI","distance":16.2,"totalDistance":6278220.03,"motion":true}   |
| 2018-03-11 14:46:22 | {"hdop":1,"odometer":0.0,"type":"FRI","distance":10.12,"totalDistance":6278230.15,"motion":false,"ignition":true}                                             |
| 2018-03-11 14:47:22 | {"hdop":1,"odometer":0.0,"type":"FRI","distance":171.57,"totalDistance":6278391.6,"motion":true,"ignition":true}                                              |
| 2018-03-11 14:47:26 | {"hdop":1,"odometer":443000.0,"batteryLevel":100,"ignition":true,"input":1,"output":0,"type":"FRI","distance":184.37,"totalDistance":6278404.4,"motion":true} |
+---------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------+
4 rows in set (0.01 sec)
```

**Comment by tananaev** [comment]
The problem should be fixed now.
